GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -846,29 +846,45 @@ private void ShowLoading(bool showSpinner = true)
         }
 
         /// <summary>
-        ///  Indicates whether the revision grid can be refreshed, i.e. it is not currently being refreshed
-        ///  or it is not in a middle of reconfiguration process guarded by <see cref="SuspendRefreshRevisions"/>
+        ///  Indicates whether the revision grid can be refreshed,
+        ///  i.e. it is not in a middle of reconfiguration process guarded by <see cref="SuspendRefreshRevisions"/>
         ///  and <see cref="ResumeRefreshRevisions"/>.
         /// </summary>
-        private bool CanRefresh => !_isRefreshingRevisions && _updatingFilters == 0;
+        private bool CanRefresh => _updatingFilters == 0;
```

Comment:
_updatingFilters is a very bad name of this guard, is no longer related to filters, just the initial start or new load.
Rename to _suspendUpdateCounter or something.

The _isRefreshingRevisions check was required as some operations started multiple refresh operations even if just one refresh was ordered. Many of these scenarios were eliminated in the update primarily Russkie did one or two years ago (I was involved too). Maybe loading a new module is the only valid scenario where this applies, maybe that handling can be removed too. Have you checked that?

Reply:
I will rename the unchanged variable.

I have tested the cancellation on switching to another repo a few times.

Reply:
Switch repo still has a guard. The potential problem is all other situations where refresh is called.
When I looked at it, I did not dare to change.

Reply:
I have checked (most of) the reload chain and am pretty confident.

E.g. quickly switch <kbd>Alt+V</kbd><kbd>O</kbd> (topo order) with Before and After: bad UX vs. just works.

Reply:
This could work